weather/weather.go

```diff
@@ -24,20 +25,30 @@ type Parser struct {
 	solarRadiation *prometheus.GaugeVec
 	rainIn         *prometheus.GaugeVec
 	ultraviolet    *prometheus.GaugeVec
+	lightning      *prometheus.GaugeVec
+	stationtype    *prometheus.GaugeVec
 }
 
-func NewParser(name string, factory *promauto.Factory) *Parser {
+func NewParser(name string, prefix string, factory *promauto.Factory) *Parser {
+	metric_prefix := ""
+	if prefix != "" {
+		metric_prefix = prefix + "_"
+	}
+
 	return &Parser{
 		name:           name,
-		temperature:    newGauge(factory, "temperature", "name", "sensor"),
-		battery:        newGauge(factory, "battery", "name", "sensor"),
-		humidity:       newGauge(factory, "humidity", "name", "sensor"),
-		barometer:      newGauge(factory, "barometer", "name", "type"),
-		windDir:        newGauge(factory, "wind_dir", "name"),
-		windSpeedMph:   newGauge(factory, "wind_speed_mph", "name", "type"),
-		solarRadiation: newGauge(factory, "solar_radiation", "name"),
-		rainIn:         newGauge(factory, "rain_in", "name", "period"),
-		ultraviolet:    newGauge(factory, "ultraviolet", "name"),
+		metric_prefix:  metric_prefix,
+		temperature:    newGauge(factory, metric_prefix + "temperature", "name", "sensor"),
+		battery:        newGauge(factory, metric_prefix + "battery", "name", "sensor"),
+		humidity:       newGauge(factory, metric_prefix + "humidity", "name", "sensor"),
+		barometer:      newGauge(factory, metric_prefix + "barometer", "name", "type"),
+		windDir:        newGauge(factory, metric_prefix + "wind_dir", "name", "period"),
+		windSpeedMph:   newGauge(factory, metric_prefix + "wind_speed_mph", "name", "type"),
+		solarRadiation: newGauge(factory, metric_prefix + "solar_radiation", "name"),
+		rainIn:         newGauge(factory, metric_prefix + "rain_in", "name", "period"),
+		ultraviolet:    newGauge(factory, metric_prefix + "ultraviolet", "name"),
+		lightning:      newGauge(factory, metric_prefix + "lightning", "name", "period", "type"),
+		stationtype:    newGauge(factory, metric_prefix + "stationtype_info", "name", "type"),
```

Comment:
I would use 3 metrics here:
- `last_lightning_strike` with label `name`
- `lightning_strikes` with labels `name` and `period` (day/hour) (no `type` label)
- `lightning_distance_miles` with label `name` (no `type` label)